database/database.go

```diff
@@ -397,18 +397,24 @@ func (db *Database) GetAllInstances() (instances []launcher.InstanceInfo, err er
 
 // GetInstanceIDs returns instance ids by filter.
 func (db *Database) GetInstanceIDs(filter cloudprotocol.InstanceFilter) (instances []string, err error) {
-	var subjectFiler, instanceFiler string
+	var serviceFilter, subjectFiler, instanceFiler, operationFilter string
```

Comment:
over complicated and easy to make mistake with spaces. Consider the following implementation:

```go

var conditionList []string;

if filter.ServiceID != nil {
    conditionList = append(conditionList, fmt.Sprintf(`serviceID = "%s"`, *filter.ServiceID)
}

...

instanceInfos, err := db.getInstancesFromQuery(fmt.Sprintf("SELECT * FROM instances WHERE %s", strings.Join(conditionList, " AND "))

Reply:
done